BL 11952 Update lambda event trigger to be a SQS Event

davidmck95 opened the pull request:
Includes logic changes as the hvt-record-handler lambda has already checked if availability has changed before publishing to email SNS topic, which means that we don't need to check this here anymore.

Commit:
Change event source from db stream to SQS event

- Remove checks around if availability has changes as the record
handler now checks this and only pushes to the queue when it's updated
- Updated tests
- New event to invoke event with
- Closes ticket #BL11952

davidmck95 commented on src/lib/email.ts:
```diff
@@ -14,8 +14,6 @@ export const buildEmailBody = (params: BuildEmailBodyParams): string => {
   const linkTemplateTag = availability.isAvailable
     ? 'link'
     : 'link';
-  const tokenKey = availability.isAvailable ? 'no' : 'yes';
```
this was not being used so I removed

sdh100shaun commented on src/handler/index.ts:
```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
     let availabilityData: AvailabilityChangeData;
     try {
-      availabilityData = extractAvailabilityData(record);
+      availabilityData = validateAvailabilityData(message);
```
As per discussion - shall we change this ? and consolidate the error handling

AnnaDodson replied:
Consolidated the try catch and handled the different errors by checking the type of error thrown

sdh100shaun commented on src/handler/index.ts:
```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
-      oldAvailability,
-      newAvailability,
-    } = availabilityData;
+    const { availability } = availabilityData;
```
should we expand the destructuring?

AnnaDodson replied:
I've got rid of the destrucing on availability as none of the others were getting deconstructed just to be set again into the object for `buildSQSMessage` to handle

AnnaDodson committed:
Refactor try catch and code tidy

- Address code review comments, tidy up some variables and comments
that weren't required
- Consolidate two try catch blocks into one and handle different
type of errors thrown

sdh100shaun committed:
Merge branch 'main' into feature/BL-11952-queue-event
